File: WWF/WF_WCF_Samples v4.0Beta1/WCF/Basic/Discovery/Basic/VB/service/Program.vb
'----------------------------------------------------------------
' Copyright (c) Microsoft Corporation.  All rights reserved.
'----------------------------------------------------------------

Imports Microsoft.VisualBasic
Imports System
Imports System.ServiceModel
Imports System.ServiceModel.Discovery

Namespace Microsoft.Samples.Discovery

	Friend Class CalculatorServiceHost
		Public Shared Sub Main()
			Dim baseAddress As New Uri("http://localhost:8000/" & Guid.NewGuid().ToString())

			Using serviceHost As New ServiceHost(GetType(CalculatorService), baseAddress)
				serviceHost.AddServiceEndpoint(GetType(ICalculatorService), New WSHttpBinding(), String.Empty)

				' Make the service discoverable over UDP multicast
				serviceHost.Description.Behaviors.Add(New ServiceDiscoveryBehavior())
				serviceHost.AddServiceEndpoint(New UdpDiscoveryEndpoint())

				serviceHost.Open()

				Console.WriteLine("Calculator Service started at {0}", baseAddress)
				Console.WriteLine()
				Console.WriteLine("Press <ENTER> to terminate the service.")
				Console.WriteLine()
				Console.ReadLine()
			End Using
		End Sub

	End Class
End Namespace

File: WWF/WF_WCF_Samples v4.0Beta1/WCF/Basic/Discovery/Basic/CS/service/Program.cs
//----------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
//----------------------------------------------------------------

namespace Microsoft.Samples.Discovery
{
    using System;
    using System.ServiceModel;
    using System.ServiceModel.Discovery;

    class CalculatorServiceHost
    {
        public static void Main()
        {
            Uri baseAddress = new Uri("http://localhost:8000/" + Guid.NewGuid().ToString());                        

            using (ServiceHost serviceHost = new ServiceHost(typeof(CalculatorService), baseAddress))
            {
                serviceHost.AddServiceEndpoint(typeof(ICalculatorService), new WSHttpBinding(), String.Empty);

                // Make the service discoverable over UDP multicast
                serviceHost.Description.Behaviors.Add(new ServiceDiscoveryBehavior());                
                serviceHost.AddServiceEndpoint(new UdpDiscoveryEndpoint());

                serviceHost.Open();

                Console.WriteLine("Calculator Service started at {0}", baseAddress);
                Console.WriteLine();
                Console.WriteLine("Press <ENTER> to terminate the service.");
                Console.WriteLine();
                Console.ReadLine();
            }
        }

    }
}
